Add "Configuration and environments" section (#7)
boostercloud/docs-site: source/includes/_configuration-and-environments.md

# Configuration and environments
One of the goals of booster is to avoid configuration parameters as much as
possible, inferring everything automatically from your code. However, there
are some aspects that can't be inferred (like the application name), or it
is useful to change them under certain circumstances, especially when using
different [environments](#environments)

> ## Booster configuration

Reviewer: ```suggestion
## Booster configuration

Booster configuration can be in any typescript file included in your project, but the recommendation is to keep a `config` folder in your project and have one file for each environment (i.e: You can have a file named `config/production.ts` file with the configuration for your production environment and a file named `config/development.ts` with a shared development environment). [The project generator](_01-your-first-backend-in-10-minutes.md#1-create-the-project) puts configuration files in the `config` folder by default.
```

> ## Booster configuration

Reviewer: I'd add some clarification before throwing the code example:

```suggestion
## Booster configuration

Booster supports multiple configuration environments (i.e: You can have a file named `src/config/production.ts` file with the configuration for your production environment and a file named `src/config/pre-production.ts` with a shared stage for QA). Let's see an example:
```

Author: Ok, but I would change this a bit to leave clear that the name of the file is not necessarily the name of the environment, but the name you pass to the first parameter. I will improve the doc to leave this distinction clear

```typescript
import { Booster } from '@boostercloud/framework-core'
import { BoosterConfig } from '@boostercloud/framework-types'
import { Provider } from '@boostercloud/framework-provider-aws'

Booster.configure('pre-production', (config: BoosterConfig): void => {
  config.appName = 'my-app-name'
  config.provider = Provider
})
```

> You configure your Booster application by calling the `Booster.configure()` method.
> Typically, this call is done in a file placed inside "src/config" folder. If you have
> used the `boost new:project` generator, you will already have this file there.

Reviewer: ```suggestion
You configure your application by calling to the `Booster.configure()` method. 
There are no restrictions about where you should do this call, but the convention is to have your configuration files in the `src/config` folder.
If you used the project generator (`boost new:project <project-name>`), this is where the config files are by default.
```

The following is the list of the fields you can configure:

- **appName:** This is the name that identifies your application. It will be used
for many things, such us prefixing the resources created by the provider. There
are certain restrictions regarding the characters you can use: all of them must be
lower-cased and can't contain spaces.
Two apps with different names are completely independent.

> - **provider:** This field contains the provider you will be using when deploying

Reviewer: ```suggestion
- **provider:** This field contains the provider library instance that Booster will use when deploying```

> or running your application. So far, there is only one provider created for Booster,

Reviewer: ```suggestion
or running your application. So far, there is only one provider supported in Booster yet,
```

`@boostercloud/framework-provider-aws`, and it is probably the one you have already
set if you used the generator to create your project. There will be more providers,
like one that allows running your Booster app locally.

## Environments

> 

Reviewer: ```suggestion

You can configure multiple environments calling to the `Booster.configure` function several times using different environment names:

```

Author: Take into account that the reason why the code-blocks are first is because of Slate, so people reading the docs will see the text first and then, on the right, they will see the code ;-)

```typescript
import { Booster } from '@boostercloud/framework-core'
import { BoosterConfig } from '@boostercloud/framework-types'
// A provider that deploys your app to AWS:
import { AWSProvider } from '@boostercloud/framework-provider-aws'
// A provider that deploys your app locally:
import { LocalProvider } from '@boostercloud/framework-provider-local' 

Booster.configure('dev', (config: BoosterConfig): void => {
  config.appName = 'fruit-store-qa'
  config.provider = LocalProvider
})

Booster.configure('stage', (config: BoosterConfig): void => {
  config.appName = 'fruit-store-stage'
  config.provider = AWSProvider
})

Booster.configure('prod', (config: BoosterConfig): void => {
  config.appName = 'fruit-store-prod'
  config.provider = AWSProvider
})

// This other configuration could be in another file that Pepe doesn't commit
Booster.configure('pepe', (config: BoosterConfig): void => {
  config.appName = 'pepe-fruit-store'
  config.provider = AWSProvider
})
```

> When you configure your app by calling `Booster.configure`, the first parameter you need to pass is the
> **environment name**, and it identifies a specific set of configuration parameters. When you deploy your 
> application, you need to specify which environment you want to deploy.

Reviewer: ```suggestion
The environment name will be required by any command from the Booster CLI that depends on the provider. 
For instance, when you deploy your application, you'll need to specify which environment you want to deploy.
```
